WorkloadManagementSystem/Agent/SiteDirector.py

```diff
@@ -341,67 +354,208 @@ def getQueues( self, resourceDict ):
           self.queueDict[queueName]['QueueName'] = queue
           self.queueDict[queueName]['Platform'] = platform
-          self.queueDict[queueName]['QueryCEFlag'] = ceDict.get( 'QueryCEFlag', "false" )
+          self.queueDict[queueName]['QueryCEFlag'] = ceDict.get('QueryCEFlag', "false")
 
           result = self.queueDict[queueName]['CE'].isValid()
           if not result['OK']:
-            self.log.fatal( result['Message'] )
+            self.log.fatal(result['Message'])
             return result
           if 'BundleProxy' in self.queueDict[queueName]['ParametersDict']:
-            if self.queueDict[queueName]['ParametersDict']['BundleProxy'].lower() in ['true','yes','1']:
+            if self.queueDict[queueName]['ParametersDict']['BundleProxy'].lower() in ['true', 'yes', '1']:
               self.queueDict[queueName]['BundleProxy'] = True
           elif 'BundleProxy' in ceDict:
-            if ceDict['BundleProxy'].lower() in ['true','yes','1']:
+            if ceDict['BundleProxy'].lower() in ['true', 'yes', '1']:
               self.queueDict[queueName]['BundleProxy'] = True
 
           if site not in self.sites:
-            self.sites.append( site )
+            self.sites.append(site)
 
           if "WholeNode" in self.queueDict[queueName]['ParametersDict']:
             self.globalParameters['WholeNode'] = 'True'
-          for parameter in [ 'MaxRAM', 'NumberOfProcessors' ]:
+          for parameter in ['MaxRAM', 'NumberOfProcessors']:
             if parameter in self.queueDict[queueName]['ParametersDict']:
-              self.globalParameters[parameter] = max( self.globalParameters[parameter],
-                                                      int( self.queueDict[queueName]['ParametersDict'][parameter] ))
-
+              self.globalParameters[parameter] = max(self.globalParameters[parameter],
+                                                     int(self.queueDict[queueName]['ParametersDict'][parameter]))
 
     return S_OK()
 
-  def execute( self ):
+  def execute(self):
     """ Main execution method (what is called at each agent cycle).
 
         It basically just calls self.submitJobs() method
     """
 
     if not self.queueDict:
-      self.log.warn( 'No site defined, exiting the cycle' )
+      self.log.warn('No site defined, exiting the cycle')
       return S_OK()
 
+    # get list of usable sites within this cycle
+    result = self.siteClient.getUsableSites()
+    if not result['OK']:
+      return result
+    self.siteMaskList = result.get('Value', [])
+
+    if self.rssFlag:
+      ceNamesList = [queue['CEName'] for queue in self.queueDict.itervalues()]
+      result = self.rssClient.getElementStatus(ceNamesList, "ComputingElement")
+      if not result['OK']:
+        self.log.error("Can not get the status of computing elements",
+                       " %s" % result['Message'])
+      self.ceMaskList = [ceName for ceName in result['Value'].iterkeys() if result['Value'][ceName]
```

Comment:
You will reach this point even if the result is S_ERROR, which will give an exception. Maybe you want a `break/ return/continue` in the error above ?

Reply:
Oh thanks!